tests/test_spatial_data.py

```diff
@@ -0,0 +1,111 @@
+import unittest
+import os
+
+from geoalchemy2 import WKBElement
+from sqlalchemy import func, event
+from sqlalchemy.schema import DropSchema
+from testing.postgresql import Postgresql
+
+from pepys_import.core.store.data_store import DataStore
+from pepys_import.core.store.db_base import base_postgres
+
+FILE_PATH = os.path.dirname(__file__)
+TEST_DATA_PATH = os.path.join(FILE_PATH, "sample_data", "csv_files")
+
+
+class SpatialDataSpatialiteTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.store = DataStore("", "", "", 0, ":memory:", db_type="sqlite")
+        self.store.initialise()
+        self.store.populate_reference(TEST_DATA_PATH)
+        self.store.populate_metadata(TEST_DATA_PATH)
+        self.store.populate_measurement(TEST_DATA_PATH)
+
+    def tearDown(self) -> None:
+        pass
+
+    def test_location(self):
+        with self.store.session_scope() as session:
+            # Filter state object by spatial location
+            first_state = (
+                self.store.session.query(self.store.db_classes.State)
+                .filter(
+                    self.store.db_classes.State.location.ST_Contains(
+                        WKBElement("POINT(46.000 32.000)")
```

Comment:
@BarisSari - I changed the `46` above to `48`, and the test still passed. I think the logic may not work in the way we expect it to.  It's worth us learning about it.  

Could you introduce a test where we search for a point we know isn't present, and verify that `None` (or similar) is returned?

Reply:
I added tests. The problem was filtering. There are two options in [Geo Alchemy tutorials](https://geoalchemy-2.readthedocs.io/en/latest/spatialite_tutorial.html). I played with them more and realized that I confused [WKBE and WKTE Elements](https://geoalchemy-2.readthedocs.io/en/latest/elements.html#geoalchemy2.elements.WKTElement) and therefore it didn't filter State objects. So, I changed tests according to that. It is not possible now to return more than one object (because one() used instead of first() after filtering). And The returned object is correctly filtered now. I added tests for non-existing points as well.

https://github.com/debrief/pepys-import/pull/43/commits/bc4e9b5534ca022f3b8130c9bb1f03aba1aa46be